source/agora/cli/multi/main.d

```diff
@@ -0,0 +1,251 @@
+/*******************************************************************************
+
+    Entry point for the Agora multi
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.cli.multi.main;
+
+import agora.api.FullNode;
+
+import agora.common.Config;
+import agora.node.Node;
+import agora.utils.Log;
+
+import vibe.core.core;
+import vibe.http.router;
+import vibe.http.server;
+import vibe.web.rest;
+
+import ocean.util.log.Logger;
+
+import std.file;
+import std.format;
+import std.getopt;
+import std.path;
+import std.stdio;
+
+import core.thread;
+
+mixin AddLogger!();
+
+/// Application entry point
+private int main (string[] args)
+{
+    if (args.length < 3)
+    {
+        writeln("usage: agora-multi path file1.yaml file2.yaml file3.yaml ...");
+    }
+
+    string path = args[1];
+    string[] files;
+
+    foreach (idx; 2..args.length)
+        files ~= buildNormalizedPath(path, args[idx]);
+
+    NodeLauncher[] launchers;
+
+    foreach (ref file; files)
+    {
+        try
+        {
+            CommandLine cmdln;
+            cmdln.config_path = file;
+            auto config = parseConfigFile(cmdln);
+
+            auto launcher = new NodeLauncher();
+
+            //launcher.startThread(config);
+            launcher.startTask(config);
+
+            launchers ~= launcher;
+
+            testConection(format("http://%s:%d", config.node.address, config.node.port));
```

Comment:
Why are we spawning the nodes and the client in the same process ? The issue is only about spawning clients, not nodes.

Reply:
I used it for testing purposes.
However, if the server port does not open, it outputs an error.
So can we get rid of `testConnection`?

Reply:
> So can we get rid of `testConnection`?

You should be able to use the client part of `test/system/` with a single-binary network.

Reply:
**In the case of starting the node**
`agora-multi start node00.yaml node01.yaml node02.yaml`

**In the case of connecting to the node**
`agora-multi test node00.yaml node01.yaml node02.yaml`

Reply:
Yes, I just added it

Reply:
What's the purpose of the `test` command ? That is not the issue.
The idea of `agora-multi` is to spawn a full network in one binary. The testing should still be separately done, by a client that may request one or more node, but it's orthogonal.

Reply:
The test command ensures that the server waits when it is running normally.

Reply:
To shut down a node, we just send it a `SIGTERM`. That's the standard way to do it on POSIX.

Reply:
I used two SIGTERM and SIGKILL, but I deleted SIGKILL now.